packages/api/tests/endpoints/workflows.js

```diff
@@ -73,3 +107,70 @@ test('CUMULUS-912 GET with pathParameters and with an unauthorized user returns
     assertions.isUnauthorizedUserResponse(t, response);
   });
 });
+
+test.serial('GET with no path parameters and an authorized user returns a list of workflows', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 200);
+    const results = JSON.parse(response.body);
+
+    t.deepEqual(results, workflowList);
+  });
+});
+
+test.serial('GET an existing workflow with an authorized user returns a specific workflow', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    pathParameters: {
+      name: 'HelloWorldWorkflow'
+    },
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 200);
+    const result = JSON.parse(response.body);
+
+    t.deepEqual(result, workflowList[0]);
+    t.is(result.template, 's3://bucket/cumulus/workflows/HelloWorldWorkflow.json');
+  });
+});
+
+test.serial('GET with path parameters returns a 404 for a nonexistent workflow', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    pathParameters: {
+      name: 'NonexistentWorkflow'
+    },
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 404);
+
+    const result = JSON.parse(response.body);
+    t.is(result.message, 'The specified key does not exist.');
+  });
+});
+
+test.serial('GET /good-workflow returns a 500 if the workflows list cannot be fetched from S3', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    pathParameters: {
+      name: 'HelloWorldWorkflow'
+    },
+    headers: authHeaders
+  };
+
+  const realBucket = process.env.bucket;
+  process.env.bucket = 'bucket-does-not-exist';
```

Comment:
The rest of this test should be wrapped in a `try/finally` block to make sure that the bucket environment variable is always reset back to its original value.

Reply:
Didn't we try that already? And the tests wasn't executing with the changed bucket? 
We had been putting the `process.env.bucket = realBucket;` back in the finally but we moved it into the return